### PR TITLE
Add connector token approval script

### PR DESCRIPTION
**Pull request by CharlieMc0**
Adds small script to approve the connector contract to spend Zeta Tokens on behalf of the signer.

### CODE REVIEW

File: packages/protocol-contracts/scripts/token-approval.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { getAddress, isNetworkName } from "@zetachain/addresses";
+import { ethers, network } from "hardhat";
+
+import { isEthNetworkName } from "../lib/contracts.helpers";
+import { ZetaEth__factory as ZetaEthFactory, ZetaNonEth__factory as ZetaNonEthFactory } from "../typechain-types";
+
+const approvalAmount = "10000000000000000000000000"; // 10000000 ZETA
```
Comment (lucas-janon) [inline packages/protocol-contracts/scripts/token-approval.ts:7]: Let's use `ethers.utils.parseEther`

File: packages/protocol-contracts/scripts/token-approval.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  let contract;
+  let factory;
+
+  if (isEthNetworkName(network.name)) {
+    factory = (await ethers.getContractFactory("ZetaEth")) as ZetaEthFactory;
+  } else {
+    factory = (await ethers.getContractFactory("ZetaNonEth")) as ZetaNonEthFactory;
+  }
+
+  contract = factory.attach(getAddress("zetaToken"));
```
Comment (lucas-janon) [inline packages/protocol-contracts/scripts/token-approval.ts:23]: Let's add shared functions (you can pretty much copy `getZetaConnectorEth` and `getZetaConnectorNonEth`)

File: packages/protocol-contracts/lib/contracts.helpers.ts [hunk 3]
```diff
@@ -101,6 +101,17 @@ export const getZetaConnectorNonEth = async (params: GetContractParams<ZetaConne
...
+  await getContract<ZetaNonEthFactory, ZetaNonEth>({
+    contractName: "ZetaNonEth",
+    ...params,
+  });
+
+export const getZetaFactoryEth = async (params: GetContractParams<ZetaEthFactory>) =>
+  await getContract<ZetaEthFactory, ZetaEth>({
+    contractName: "ZetaNonEth",
+    ...params,
+  });
```
Comment (lucas-janon) [inline packages/protocol-contracts/lib/contracts.helpers.ts:114]:
```suggestion
  });

```

**Review by lucas-janon — APPROVED** [review]
LGTM

**Review by brewmaster012 — APPROVED** [review]
LGTM

**Pull request merged**